CPP_STYLE_GUIDE.md

```diff
@@ -1,5 +1,22 @@
 # C++ Style Guide
 
+## Table of Contents
+
+* [Left-leaning (C++ style) asterisks for pointer declarations](#left-leaning-c-style-asterisks-for-pointer-declarations)
```

Comment:
Can we group the rules into different categories? For what we have here I think we should at least have "formatting" and "memory management".

Reply:
Sure. Is it OK to do that in a separate PR?

Reply:
@fhinkel Of course!